.github/workflows/tests.yml

```diff
@@ -0,0 +1,45 @@
+name: Tests
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache pip
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+        restore-keys: |
+          ${{ runner.os }}-pip
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools wheel nox 'poetry==1.2.0.a2' poetry-export-plugin
+    - name: Cache poetry
+      uses: actions/cache@v1
```

Comment:
```suggestion
      uses: actions/cache@v2.1.7
```

Reply:
Why do we need a specific version and not just v2?

Reply:
not exactly specific, just suggesting latest stable release